tests/container_test.rs

```diff
@@ -280,6 +283,65 @@ async fn kill_container_test(docker: Docker) -> Result<(), Error> {
     Ok(())
 }
 
+async fn attach_container_test(docker: Docker) -> Result<(), Error> {
+    create_daemon(&docker, "integration_test_attach_container").await?;
+
+    let AttachContainerResults {
+        mut output,
+        mut input,
+    } = docker
+        .attach_container(
+            "integration_test_attach_container",
+            Some(AttachContainerOptions::<String> {
+                stream: Some(true),
+                stdout: Some(true),
+                stdin: Some(true),
+                ..Default::default()
+            }),
+        )
+        .await?;
+
+    input.write("hello!\n".as_bytes()).await?;
+
+    let result = match output.next().await {
+        Some(Ok(result)) => Some(result.to_string()),
+        _ => None,
+    };
+
+    let _ = &docker
+        .kill_container(
+            "integration_test_attach_container",
+            None::<KillContainerOptions<String>>,
+        )
+        .await?;
+
+    let _ = &docker
+        .wait_container(
+            "integration_test_attach_container",
+            None::<WaitContainerOptions<String>>,
+        )
+        .try_collect::<Vec<_>>()
+        .await?;
+
+    let _ = &docker
+        .remove_container(
+            "integration_test_attach_container",
+            None::<RemoveContainerOptions>,
+        )
+        .await?;
+
+    if cfg!(windows) {
+        assert_eq!(
+            result,
+            Some("\u{1b}[2J\u{1b}[H\u{1b}[37;40m\r\n".to_string())
```

Comment:
hmm... looks strange, do you have any idea why it's saying something other than the *nix codepath?

Reply:
@fussybeaver no clue, but I believe it is an ANSI code for clearing the terminal if I'm not mistaken. It was a bit of a pain to debug as I was only able to reproduce it in the CI.

I'm happy to investigate it more if you deem needed, however would be great to reproduce it somehow on my machine. Any tips?

Reply:
It'd be best to get this running locally to check what is going on. Last time I got this working on an evaluation version of VMware workstation (which is needed because of the Hyper-V restrictions in Windows Docker) and a Windows eval image.

Reply:
Thanks. I will take a look.

Reply:
So yeah the ANSI value was from what the windows terminal does when it instantiates:
```
^[[2J^[[H^[[1H^[[37;40m^[[K^[[2H^[[K^[[3H^[[K^[[4H^[[K^[[5H^[[K^[[6H^[[K^[[7H^[[K^[[8H^[[K^[[9H^[[K^[[10H^[[K^[[11H^[[K^[[12H^[[K^[[13H^[[K^[[14H^[[K^[[15H^[[K^[[16H^[[K^[[17H^[[K^[[18H^[[K^[[19H^[[K^[[20H^[[K^[[21H^[[K^[[22H^[[K^[[23H^[[K^[[24H^[[K^[[25H^[[K^[[1;1HMicrosoft Windows [Version 10.0.14393]
(c) 2016 Microsoft Corporation. All rights reserved.
```

I implemented the test now so that it searches for the input from the output stream which works for all OS.

@fussybeaver would you mind rerunning the `test_http` test as it looks like it got a temporary error?